### PR TITLE
Change sample to example in TextVectorization

### LINKED ISSUE #48298: Sample or example?
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/layers/experimental/preprocessing/TextVectorization

### Clear description

Lots of the description use the word _sample_. However, [Google Machine Learning Glossary](https://developers.google.com/machine-learning/glossary) does not have an entry of _sample_. It does have _example_:

> example
> One row of a dataset. An example contains one or more features and possibly a label. See also labeled example and unlabeled example.

Since this class works on dataset, I think the description is talking about examples.

Before we have samples, there must be a process of sampling. I don't think TextVectorization is doing sampling. Therefore it's unproper to call examples samples.

Should "sample" be replaced by "example"?

### PR DESCRIPTION
**Pull request by gqqnbig**
I made an issue about this, and Mark Daoust says "Example" would be a slightly better fit. 

Therefore I changed all "sample" to "example" in the docstr of TextVectorization.

Fix #48298

### CODE REVIEW

**Review by Nirzak — APPROVED** [review]
Approved it as daoust said so. But I think on some instances sample would be better.

**Review by rthadur — APPROVED** [review]

**Comment by gbaned** [comment]
@gqqnbig  Can you please address Ubuntu Sanity errors? Thanks!

**Comment by gqqnbig** [comment]
@gbaned should be fixed. Please check.

**Review by gbaned — APPROVED** [review]
